### PR TITLE
Value date and currency for invoices, invoice print format for customers

### PR DESCRIPTION
**Pull request by ajmyyra**
Hi there,

Two small incremental changes here:
- Allow specifying a value date for invoice, as it can differ from actual invoicing date.
- Allow specifying invoice print channel format for customers. This allows for international invoices to be printed without the bank transfer details usually included at the bottom of the invoice.

Both changes are included in test data, and tested against the Netvisor API.

Thank you!

### CODE REVIEW

**Comment by ajmyyra** [comment]
Forgot to add the third incremental feature: allowing to specify currencies when making invoices. IF this isn't specified, your own default currency is used, but you might want to send invoices with other currencies as well, so it's good to support it.

File: netvisor_api_client/schemas/sales_invoices/create.py [hunk 1]
```diff
@@ -98,9 +98,11 @@ def __setattr__(self, attr, value):
 class CreateSalesInvoiceSchema(RejectUnknownFieldsSchema):
     sales_invoice_number = fields.Integer(attribute='number')
     sales_invoice_date = fields.Date(attribute='date')
+    sales_invoice_value_date = fields.Date(attribute='value_date')
     sales_invoice_delivery_date = fields.Date(attribute='delivery_date')
     sales_invoice_reference_number = fields.String(attribute='reference_number')
     sales_invoice_amount = Decimal(attribute='amount')
+    amount_currency = fields.String(attribute='currency')
```
Comment (lekksi) [inline netvisor_api_client/schemas/sales_invoices/create.py]: For uniformity

```suggestion
    sales_invoice_amount_currency = fields.String(attribute='currency')
```

File: netvisor_api_client/schemas/sales_invoices/create.py [hunk 2]
```diff
@@ -158,6 +160,13 @@ def post_dump(self, data):
                 '@type': 'netvisor'
             }
 
+        if 'amount_currency' in data:
```
Comment (lekksi) [inline netvisor_api_client/schemas/sales_invoices/create.py]: For uniformity

```suggestion
        if 'sales_invoice_amount_currency' in data:
```

File: netvisor_api_client/schemas/sales_invoices/create.py [hunk 3]
```diff
@@ -158,6 +160,13 @@ def post_dump(self, data):
...
+            data['sales_invoice_amount'] = {
+                '#text': data['sales_invoice_amount'],
+                '@iso4217currencycode': data['amount_currency']
```
Comment (lekksi) [inline netvisor_api_client/schemas/sales_invoices/create.py]:
```suggestion
                '@iso4217currencycode': data['sales_invoice_amount_currency']
```

File: netvisor_api_client/schemas/sales_invoices/create.py [hunk 4]
```diff
@@ -158,6 +160,13 @@ def post_dump(self, data):
...
+            }
+            del data['amount_currency']
```
Comment (lekksi) [inline netvisor_api_client/schemas/sales_invoices/create.py]:
```suggestion
            del data['sales_invoice_amount_currency']
```

**Review by lekksi — CHANGES_REQUESTED** [review]
Thank you for the PR, good stuff! 
Made one suggestion for improved uniformity, with that ready to go!

**Comment by ajmyyra** [comment]
Thank you! I’ve updated the suggestions now, they looked good! Please tell me if I should squash the changes to the original commit, and I can do that before merging.

**Review by lekksi — APPROVED** [review]

**Comment by lekksi** [comment]
Thanks again for the PR! It's now released at PyPI.

**Pull request merged**